src/plugins/intel_gpu/src/graph/impls/onednn/lstm_seq_onednn.hpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "lstm_seq_inst.h"
+#include "reshape_inst.h"
+#include "intel_gpu/runtime/utils.hpp"
+#include "impls/registry/implementation_manager.hpp"
+#include "transformations/utils/utils.hpp"
+#include "impls/onednn/utils.hpp"
+
+#include <memory>
+
+
+namespace cldnn {
+namespace onednn {
+
+struct LSTMSeqImplementationManager : public ImplementationManager {
+    OV_GPU_PRIMITIVE_IMPL("onednn::lstm_seq")
+    LSTMSeqImplementationManager(shape_types shape_type) : ImplementationManager(impl_types::onednn, shape_type) {}
+    std::unique_ptr<primitive_impl> create_impl(const program_node& node, const kernel_impl_params& params) const override;
+
+    bool validate_impl(const program_node& node) const override {
+        assert(node.is_type<lstm_seq>());
+        const auto& info = node.get_program().get_engine().get_device_info();
+        if (info.arch == gpu_arch::unknown)
+            return false;
+
+        const auto& lstm_seq_node = node.as<lstm_seq>();
+        const auto& in_layout = lstm_seq_node.get_input_layout(0);
+        const auto& out_layout = lstm_seq_node.get_output_layout(0);
+
+        if (node.get_input_layout(0).format != cldnn::format::bfyx && node.get_input_layout(0).format != cldnn::format::fbyx
+            && node.get_input_layout(0).format != cldnn::format::ybfx)
+            return false;
+
+        if (!is_supported_pad(in_layout) || !is_supported_pad(out_layout))
+            return false;
+
+        auto in0_dt = node.get_input_layout(0).data_type;
+        auto in1_dt = node.get_input_layout(1).data_type;
+        auto in2_dt = node.get_input_layout(2).data_type;
+        auto in3_dt = node.get_input_layout(3).data_type;
+        auto in4_dt = node.get_input_layout(4).data_type;
+        auto in5_dt = node.get_input_layout(5).data_type;
+        auto out0_dt = node.get_output_layout(0).data_type;
+        auto out1_dt = node.get_output_layout(1).data_type;
+        auto out2_dt = node.get_output_layout(2).data_type;
+        bool cell_state_check = one_of(in2_dt, {data_types::f16, data_types::bf16, data_types::f32}) &&
+            one_of(out2_dt, {data_types::f16, data_types::bf16, data_types::f32});
+        bool f16_case = everyone_is(data_types::f16, in0_dt, in1_dt, in3_dt, in4_dt, out0_dt, out1_dt);
+        bool bf16_case = everyone_is(data_types::bf16, in0_dt, in1_dt, in3_dt, in4_dt, out0_dt, out1_dt);
```

Comment:
bf16 is not supported by GPU plugin for now. I think it can be removed from here as well

Reply:
done